pkg/actions/addon/create.go

```diff
@@ -176,8 +177,19 @@ func (a *Manager) getRecommendedPolicies(addon *api.Addon) []string {
 	}
 }
 
-func (a *Manager) createRoleUsingAttachPolicyARNs(addon *api.Addon) (string, error) {
-	resourceSet := builder.NewIAMRoleResourceSetWithAttachPolicyARNs(addon.Name, addon.AttachPolicyARNs, a.oidcManager)
+func (a *Manager) getKnownServiceAccountLocation(addon *api.Addon) (string, string) {
+	// API isn't case sensitive
+	switch strings.ToLower(addon.Name) {
```

Comment:
random thought, addons could have `Name() string` which auto lowercases the name

Reply:
great idea! I'll update

Reply:
I ended up going back a forth a bit on this (the smallest thing suddenly becomes so annoying!±). We can't add a `Name()` func because it clashes with the variable `Name`. I didn't want to change the variable Name or make it private. I was left with two alternatives:
- Mutate the `Name` at some point early on, e.g when loading config, to ensure its always lower cased
- Have a different name for the func, `NameLowerCase()`

The first introduces more hidden logic away from the core action on addons, the later keeps it a bit closer so I went with that.

Reply:
All not a big deal indeed, although i would have just made it private since we never need the upper case name.
But `NameLowerCase` isn't so descriptive maybe like `CanonicalName` or something that explains why you'd want to use the function (otherwise one might as well just use `ToLower(Name)`)